presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -314,7 +316,15 @@ else if (table.getTableType().equals(EXTERNAL_TABLE.name())) {
     @Override
     public synchronized Optional<Table> getTable(HiveIdentity identity, String databaseName, String tableName)
     {
-        return getTable(databaseName, tableName);
+        Optional<Table> optionalTable = getTable(databaseName, tableName);
```

Comment:
We shouldn't need any view specific logic in the metastore, as the metastore's job is to simply store/retrieve the table metadata. This specific check seems to be a sanity check that the file metastore doesn't contain a Hive view, but that should be impossible since the code is only used by Presto.

Reply:
got it, agreed. `ThriftHiveMetastore::getAllViews` also involves usage of view specific config, may be we can move it out too in future.

Reply:
You're right, it's always had logic to do filtering based on `presto_view` flag, but I forgot that it also has special logic based on whether or not we're translating Hive views. In that case, maybe it is required to add more logic to `ThriftHiveMetastore`. Though `FileHiveMetastore` doesn't have any today, so I'd need to be more convinced for that one.